fiftyone/server/events.py

```diff
@@ -256,18 +256,53 @@ async def _initialize_listener(payload: ListenPayload) -> InitializedListener:
         _app_count += 1
 
     current = state.dataset.name if state.dataset else None
-    if is_app and payload.initializer != current:
-        if payload.initializer:
+    current_saved_view = state.view_name
+    if not isinstance(payload.initializer, fos.StateDescription):
+        update = False
+        if (
+            payload.initializer.dataset
+            and payload.initializer.dataset != current
+        ):
+            update = True
             try:
-                state.dataset = fo.load_dataset(payload.initializer)
+                state.dataset = fo.load_dataset(payload.initializer.dataset)
+                state.selected = []
+                state.selected_labels = []
+                state.view = None
+                state.view_name = None
             except:
                 state.dataset = None
-            state.selected = []
-            state.selected_labels = []
-            state.view = None
-            state.saved_view_slug = None
-            state.changing_saved_view = False
+                state.selected = []
+                state.selected_labels = []
+                state.view = None
+                state.view_name = None
+            else:
+                if payload.initializer.view:
+                    try:
+                        state.view = state.dataset.load_saved_view(
+                            payload.initializer.view
```

Comment:
is the `payload.initializer` the url? and so would `payload.initializer.view` be the value of `view` in the search params?

Reply:
This is a dataclass. It is defined [here](https://github.com/voxel51/fiftyone/blob/43a8752b3c48e86c7c9da8e9be05173ddf084adc/fiftyone/core/session/events.py#L98)